src/todo-list/todo-list.ts

```diff
@@ -0,0 +1,350 @@
+import { TodoListRenderer } from 'src/models/todo-list.interface';
+import { TodoListAppStyles } from 'src/components/todo-list-styles.components';
+import {
+  type EventBusType,
+  type SelectBtnType,
+  type TodoDndPayload,
+  type TodoListAppOptions,
+  type TodoListItem,
+  Utils,
+} from 'src/types/todo.types';
+import { BUTTON_TYPES, DEFAULT_LABEL, EVENT_BUS_TYPES } from 'src/constants/todo-list.const';
+import { TodoListElementBuilder } from 'src/components/todo-list-element-builder.components';
+import { TodoListDnDElementBuilder } from 'src/components/todo-list-dnd-element-builder.components';
+import { AbstractTodoListHandler } from 'src/components/abstract-todo-list-handler';
+
+export default class TodoListApp extends AbstractTodoListHandler implements TodoListRenderer {
+  private _instanceId = `todo-list-${Utils.createRandomKey(8)}`;
+  private _initialLabel = DEFAULT_LABEL;
+
+  get initialLabel() {
+    return this._initialLabel;
+  }
+
+  set initialLabel(value) {
+    this._initialLabel = value;
+  }
+
+  get instanceId() {
+    return this._instanceId;
+  }
+  set instanceId(value) {
+    this._instanceId = value;
+  }
+
+  private _todoStyles!: TodoListAppStyles;
+  get todoStyles() {
+    return this._todoStyles;
+  }
+  set todoStyles(value) {
+    this._todoStyles = value;
+  }
+
+  private _builder!: TodoListElementBuilder;
+
+  get builder() {
+    return this._builder;
+  }
+  set builder(value) {
+    this._builder = value;
+  }
+
+  private _options!: TodoListAppOptions;
+
+  get options() {
+    return this._options;
+  }
+  set options(value) {
+    this._options = value;
+  }
+
+  get useDnd() {
+    return this.options.useDnd || false;
+  }
+
+  private _selectedBtn: SelectBtnType = BUTTON_TYPES.ALL_ITEMS;
+
+  get selectedBtn() {
+    return this._selectedBtn;
+  }
+
+  set selectedBtn(value) {
+    this._selectedBtn = value;
+  }
+
+  private _layouts: {
+    root: HTMLDivElement | null;
+    ul: HTMLUListElement | null;
+    input: HTMLInputElement | null;
+    buttonWrapper: HTMLDivElement | null;
+    itemCnt: HTMLLabelElement | null;
+    allItems: HTMLButtonElement | null;
+    activeItems: HTMLButtonElement | null;
+    completedItems: HTMLButtonElement | null;
+    clearCompleted: HTMLDivElement | null;
+    noItems: HTMLLIElement | null;
+    buttonPack: HTMLDivElement | null;
+  } = {
+    root: null,
+    ul: null,
+    input: null,
+    buttonWrapper: null,
+    itemCnt: null,
+    allItems: null,
+    activeItems: null,
+    completedItems: null,
+    clearCompleted: null,
+    noItems: null,
+    buttonPack: null,
+  };
```

Comment:
```suggestion
export type TodoLayoutTypes = {
    root: HTMLDivElement | null;
    ul: HTMLUListElement | null;
    ...
 }
 ...
  private _layouts: TodoLayoutTypes = {
    root: null,
    ul: null,
    ...
  };
```
타입을 따로 분리하면 좋을듯합니다.
해당 파일 길이가 너무 길어집니다ㅠ